lib/roby/custom_require.rb

```diff
@@ -4,16 +4,21 @@
 module Kernel
     alias syskit_original_require require
     def require(path)
-        # Only deal with paths starting with models/, the other ones are
-        # unambiguous
-        if path =~ /^models\//
+        # Only deal with paths which is not prefixed with an app_dir basename,
+        # the other ones are unambiguous
+
+        apps = Roby.app.search_path.map do |app_dir|
+            File.basename(app_dir)
+        end
```

Comment:
Notice that the part of the code that deals with the "apps" calls `find_registered_app_path`. I.e. it is in principle possible that the 'app name' is not the app search basedir. More likely, there are paths in seach_path that are not registered apps.

If it matches against `/^(\w+)\//`, check if the capture is a registered app name. If it is, try that registered app's base dir. If not, try the search path and finally fallback to syskit_original_require.

Also need to bypass the whole mechanism for full paths. Should apply on relative paths only.

Reply:
I suggest that you write unit tests for this too ...

Reply:
tests where added